lib/matplotlib/testing/exceptions.py

```diff
@@ -1,15 +1,3 @@
-class KnownFailureTest(Exception):
```

Comment:
Just to note, matplotlib and many other projects depend on some of numpy's
testing features. By your logic, that shouldn't be done and every project
should implement their own numerical comparison tools?

I am not exactly sure how to proceed here.

On Wed, Aug 17, 2016 at 1:31 PM, Nikita Kniazev <notifications@github.com>
wrote:

> In lib/matplotlib/testing/exceptions.py
> <https://github.com/matplotlib/matplotlib/pull/6920#discussion_r75168432>:
>
> > @@ -1,15 +1,3 @@
> > -class KnownFailureTest(Exception):
>
> I cannot agree with this for two reasons:
>
>    - PRs that relay on raise KnownFailureTest should not pass after this
>    PR ever, and this change will force them to switch on xfail
>    - Other projects must not relay on matplotlib testing features, it is
>    their problem if they are
>
> I can understand backward compatibility on matplotlib core features, but
> with removing nose support all this stuff will gone. I have already tried
> my best to make dual nose-pytest support, but leaving nose-specific stuff
> on the current places will blow something in any time in the future.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/matplotlib/matplotlib/pull/6920/files/f20efb99d67e581f18f8fd58db07d6279528fcbd#r75168432>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AARy-HYvL9iMog7FVqaAIeBGMR-v5ev-ks5qg0WIgaJpZM4Jemgm>
> .
>